crates/block-producer/src/server.rs

```diff
@@ -90,6 +97,7 @@ impl BlockProducer {
             store,
             rpc_listener,
             chain_tip,
+            remote_batch_prover,
```

Comment:
I wonder whether instead of making `remote_batch_prover` a field in `BlockProducer`, we should just pass it to the `BatchBuilder`. Specifically, we could have `BatchBuilder::new()` constructor which looks something like this:
```rust
impl BatchBuilder {
    pub fn new(prover_url: Option<Url>) -> Self {
        ...
    }
}
```
Then, how to construct the prover would be the internal concern of the `BatchBuilder`, and here, we'd just do something like:
```rust
BatchBuilder::new(config.batch_prover_url)
```